src/javascript/event.js

```diff
@@ -0,0 +1,134 @@
+  this.pagingTarget = pagingTarget;
+  this.prefix = prefix;
+  this.panel = null;
+  this.list = list;
+  this.leftArrow = null;
+  this.rightArrow = null;
+  this.indicator = null;
+}
+
+Event.prototype = {
+  init,
+  renderEvent,
+  getEventBlock,
+  renderArrow,
+  renderIndicator,
+  createArrowEvent,
+  createIndicatorEvent,
+  arrowEvent,
+  indicatorEvent,
+  movePanel
+}
+
+Event.constructor = Event;
+
+function init() {
+  this.renderEvent();
+  this.renderArrow();
+  this.renderIndicator();
+  this.createArrowEvent();
+  this.createIndicatorEvent();
+  this.arrowEvent('prev', false);
+}
+
+function renderEvent() {
+  const panel = document.createElement('div');
+  panel.classList.add('panel');
+  panel.innerHTML = this.list.reduce((acc, item) => acc += this.getEventBlock(item), '');
+  this.target.append(panel);
+  this.panel = panel;
+}
+
+function getEventBlock(item) {
+  return `<a class="panel__item"><img src="${this.prefix}${item.src}"></a>`;
+}
+
+function renderArrow() {
+  const arrowWrap = document.createElement('div');
+  arrowWrap.classList.add('paging-arrow');
+
+  const leftArrow = document.createElement('div');
+  leftArrow.classList.add('paging-arrow__left');
+  leftArrow.dataset.arrow = 'prev';
+
+  const rightArrow = document.createElement('div');
+  rightArrow.classList.add('paging-arrow__right');
+  rightArrow.dataset.arrow = 'next';
+
+  arrowWrap.append(leftArrow);
+  arrowWrap.append(rightArrow);
+  this.pagingTarget.append(arrowWrap);
+  this.leftArrow = leftArrow;
+  this.rightArrow = rightArrow;
+}
+
+function renderIndicator() {
+  const indicatorWrap = document.createElement('div');
+  indicatorWrap.classList.add('paging-indicator');
+
+  const buttons = this.list.reduce((acc, _) => acc += '<div class="paging-indicator__button"></div>', '');
+  indicatorWrap.innerHTML = buttons;
+  indicatorWrap.children[1].classList.add('active');
+
+  this.pagingTarget.append(indicatorWrap);
+  this.indicator = indicatorWrap;
+}
+
+function createArrowEvent() {
+  this.leftArrow.addEventListener('click', () => this.arrowEvent('prev'));
+  this.rightArrow.addEventListener('click', () => this.arrowEvent('next'));
+}
+
+function createIndicatorEvent() {
+  this.indicator.addEventListener('click', ({ target }) => this.indicatorEvent(target));
+}
+
+function arrowEvent(type) {
+  const toggleTransitionClassName = 'transition-on';
+  const transitionClassName = 'move-' + type;
+  const { panel } = this;
+  panel.classList.add(toggleTransitionClassName, transitionClassName);
+  setTimeout(() => {
+    this.movePanel(type);
+    panel.classList.remove(toggleTransitionClassName, transitionClassName);
+  }, 300);
+}
+
+function indicatorEvent(target) {
+  if(target.classList.contains('active')) return;
+  let activeIndex = 0;
+  let targetIndex = 0;
+  this.indicator.childNodes.forEach((elem, i) => {
+    if(elem.classList.contains('active'))
+      activeIndex = i;
+    if(elem === target)
+      targetIndex = i;
+  });
```

Comment:
`findIndex`를 활용해 보면 어떨까요?

```suggestion
const indicators = Array.from(this.indicator.childNodes).reverse();
const activeIndex = indicators.findIndex(el => el.classList.contains('active'));
const targetIndex = indicators.findIndex(el => el === target);
```

Reply:
> `findIndex`를 활용해 보면 어떨까요?

시간 복잡도에 대한 고민이 있었는데
곰곰히 생각해보니 한 번의 반복에서 분기를 타기 때문에 고려할 부분이 아니었네요!
제안해주신 코드대로 수정하도록 하겠습니다!

Reply:
네 단순히 동기적인 반복문이고 성능 이슈가 생길만 한 작업도 없어서 시간 복잡도가 n이나 3n이나 차이가 없을 것 같아요. 가독성 높은 코드를 취하는게 더 이득이라 생각합니다 ㅎㅎ